src/sonar-scanner-params.js

```diff
@@ -100,8 +101,11 @@ function defineSonarScannerParams(params, projectBaseDir, sqScannerParamsFromEnv
   if (params.serverUrl) {
     sonarScannerParams['sonar.host.url'] = params.serverUrl;
   }
+  if (params.login) {
+    sonarScannerParams['sonar.login'] = params.login;
+  }
   if (params.token) {
-    sonarScannerParams['sonar.login'] = params.token;
+    sonarScannerParams['sonar.token'] = params.token;
```

Comment:
Should we throw an error if both are used?

Reply:
We could throw an error, but I am not convinced about the value. Setting both properties won't make the analysis fail. The only thing that could happen is a warning about the deprecated property if a user is using a SonarQube v10 version.

Reply:
There will be no failure; SonarQube will just use whatever is provided:

https://github.com/SonarSource/sonarqube/blob/a157fe73363a973afe1ab7d7d8277940bfa19c08/sonar-scanner-engine/src/main/java/org/sonar/scanner/bootstrap/ScannerWsClientProvider.java#L48-L56